### PR TITLE
[NIFI-1761] Initial AngularJS application bootstrapping and breadcrumbs directive

### PR DESCRIPTION
**Pull request by scottyaslan**

### CODE REVIEW

**Comment by joewitt** [comment]
The author email appears tied to a local system name.  Probably not what you wanted.

There doesn't appear to be updates to the LICENSE/NOTICE of NiFi which will be needed given this is a source dependency.  It appears to include a lot of cruft like README files and gitignore files and so on which seem avoidable.

**Comment by mcgilman** [comment]
@scottyaslan Looks like you need to update the `apache-rat-plugin` configuration to exclude [1] the angular stuff that's been introduced. Let me know if you have any questions.

[1] https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/pom.xml#L674

**Comment by apiri** [comment]
Could we get a JIRA established (https://issues.apache.org/jira/browse/NIFI) to track these efforts?  With things that seem to be as transformative as this it would be nice to have the corresponding issue for planning as well as at release time if this is incorporated for the sake of docs and the like as well as to capture the associated history.

**Comment by mcgilman** [comment]
@scottyaslan A couple things...

- Looks there is a whitespace being inserted at the end group name in the breadcrumbs.
- The breadcrumb alignment might be off by a pixel or two.
- The bottom of the breadcrumb text is getting clipped.
- Once the breadcrumb text has overflowed and the group changes, the view does not appear to reset correctly.
- Looks like the scroll direction changed. Was this intended?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/pom.xml [hunk 1]
```diff
@@ -628,8 +634,13 @@
                                 js/d3/**/*,
                                 js/codemirror/**/*,
                                 js/jquery/**/*,
+                                js/**/**/*,
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/pom.xml]: Why was this line added? It appears to be includes all the JS resources that was aggregated above.

**Comment by mcgilman** [comment]
Ensure breadcrumb placement is correct when bottom banner is visible.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/pom.xml (hunk 1 above)
Comment (scottyaslan) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/pom.xml]: This was for the new angular libs, I will update it to be more specific: 

js/angular/**/*

**Comment by mcgilman** [comment]
Looks great! Thanks @scottyaslan
